### PR TITLE
valeronoi: init at 0.1.9

### PR DESCRIPTION
**Pull request by nova-r**
###### Description of changes

https://github.com/ccoors/Valeronoi
Valeronoi ([Valetudo](https://valetudo.cloud/) + Voronoi) is a companion for Valetudo for generating WiFi signal strength maps. It visualizes them using a [Voronoi diagram](https://en.wikipedia.org/wiki/Voronoi_diagram).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/valeronoi/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "-DCMAKE_BUILD_TYPE=Release"
+  ];
+
+  enableParallelBuilding = true;
+
+  doCheck = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/ccoors/Valeronoi/";
+    description = " A WiFi mapping companion app for Valetudo ";
```
Comment (Janik-Haag) [inline pkgs/tools/misc/valeronoi/default.nix:53]:
```suggestion
    description = "A WiFi mapping companion app for Valetudo";
```

**Review by Janik-Haag — CHANGES_REQUESTED** [review]

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 244342` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>valeronoi</li>
  </ul>
</details>

**Review by Janik-Haag — COMMENTED** [review]
in the $out/share folder are some empty folders and unused icons. You should add a desktopItem since this is a graphical application.

File: pkgs/tools/misc/valeronoi/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    wrapQtAppsHook
+    copyDesktopItems
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+  ];
+
+  enableParallelBuilding = true;
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/valeronoi/default.nix:50]:
```suggestion
```
this is already set by cmake

File: pkgs/tools/misc/valeronoi/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    boost
+    cgal
+    cmake
+    openssl
+    wrapQtAppsHook
+    copyDesktopItems
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/valeronoi/default.nix:46]: This is already set by the setup hook

File: pkgs/tools/misc/valeronoi/default.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    qtbase
+    qtimageformats
+    qtsvg
+  ];
+
+  nativeBuildInputs = [
+    boost
+    cgal
+    cmake
+    openssl
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/valeronoi/default.nix:40]: Why do we need cgal and openssl in nativeBuildInputs?

File: pkgs/tools/misc/valeronoi/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+, mpfr
+, openssl
+, qtbase
+, qtimageformats
+, qtsvg
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "Valeronoi";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/valeronoi/default.nix:18]:
```suggestion
  pname = "valeronoi";
```

File: pkgs/tools/misc/valeronoi/default.nix (hunk 4 above)
Comment (nova-r) [reply pkgs/tools/misc/valeronoi/default.nix]: I've moved cgal to buildInputs, as the package does not build without this. 

It seems that building and executing the binary still works without openssl,so i have removed it.

Comment (Janik-Haag) [reply pkgs/tools/misc/valeronoi/default.nix]: Did you test if the connection to the vacuum still works? Openssl probably won't be invoked before.

Comment (nova-r) [reply pkgs/tools/misc/valeronoi/default.nix]: I have tested this briefly and I haven't had any issues. I can still find the vacuum in my local network and connect to it. 

I'll test it further later today and let you know if everything works

Comment (nova-r) [reply pkgs/tools/misc/valeronoi/default.nix]: I've built and tested the program again on a different machine and haven't encountered any problems. Therefore, I think that valeronoi does not need openssl to function

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**